### PR TITLE
Feat 79 Initial/Followup Fields

### LINKED ISSUE #79: Map 2023 Sharepoint Initial and Followup Fields
# User story
As a user, I want duplicate fields (anaesthesia, recovery time, instrument id) to be mapped completely. These fields are procedure-agnostic in the database (with 2 backend fields each) but we want to map them to procedure-specific fields (anaesthesia, recovery time, instrument id). 

# Acceptance criteria
- [x] use fields added to Sharepoint (Burr1 Perform During, Burr2 Perform During, Burr3 Perform During, Burr4 Perform During, Headpost Perform During, Craniotomy Perform During)
- [x]  map anaesthesia duration (Iso_x0020_On, FirstInjectionIsoDuration), anaesthesia level (HPIsoLevel, Round1InjIsolevel), workstation_id (HpWorkStation, WorkStation1stInjection), recovery_time (HPRecovery, FirstInjRecovery), Nano instrument_id (NanojectNumberInj10, NanojectNumberInj2), ionto instrument_id (IontoNumberInj1, IontoNumberInj2)
- [x] map for all procedures (injections and fiber implants in burr hole methods, craniotomies, headframes)
- [x] update weight to do initial/followup
- [x] update date to do initial/followup
- [ ] update example list items in resources with new fields and update unit tests

# Sprint Ready Checklist 
- [x] 1. Acceptance criteria defined
- [x] 2. Team understands acceptance criteria
- [x] 3. Team has defined solution / steps to satisfy acceptance criteria
- [x] 4. Acceptance criteria is verifiable / testable
- [x] 5. External / 3rd Party dependencies identified
- [x] 6. Ticket is prioritized and sized

# Notes
will need to meet with sharepoint engineers to answer some of these questions

### PR DESCRIPTION
**Pull request by mekhlakapoor**
closes #79 
maps fields affected by Initial/Followup classifiers

### CODE REVIEW

**Review by jtyoung84 — COMMENTED** [review]
@saskiad @mekhlakapoor The duration fields in the pydantic models should probably be `timedelta` types, which is python's equivalent of Java's Duration class. The `time` class is supposed to represent a time of day, which has hard limits like hour < 24 etc. We can also avoid writing workarounds like the ones in the utils package here.

**Comment by mekhlakapoor** [comment]
> @saskiad @mekhlakapoor The duration fields in the pydantic models should probably be `timedelta` types, which is python's equivalent of Java's Duration class. The `time` class is supposed to represent a time of day, which has hard limits like hour < 24 etc. We can also avoid writing workarounds like the ones in the utils package here.

this is a good idea, but we may still need to write some workarounds tho because the duration fields in the SWR are in some cases ints in hours, sometimes minutes, and sometimes strings

File: src/aind_metadata_service/sharepoint/client.py [hunk 1]
```diff
@@ -98,6 +100,12 @@ class WellType(Enum):
         WHC_NP = "WHC NP"
         OTHER = "Other"
 
+    class SurgeryOrder(Enum):
+        """Enum class for Initial/FollowUp classifiers"""
+
+        INITIAL = "Initial Surgery"
+        FOLLOW_UP = "Follow up Surgery"
```
Comment (saskiad) [inline src/aind_metadata_service/sharepoint/client.py:107]: can we just call this second surgery? Follow up to me really implies like something went wrong with teh first surgery, while this really is usually just a separate, second surgery.

Comment (mekhlakapoor) [reply src/aind_metadata_service/sharepoint/client.py:107]: sure, i'll go ahead and do that

**Comment by jtyoung84** [comment]
> @saskiad @mekhlakapoor The duration fields in the pydantic models should probably be `timedelta` types, which is python's equivalent of Java's Duration class. The `time` class is supposed to represent a time of day, which has hard limits like hour < 24 etc. We can also avoid writing workarounds like the ones in the utils package here.

@mekhlakapoor It looks like we settled on keeping Durations as Minutes. I'd recommend using the timedelta class to do most of the conversion. Something like:
```
>>> timedelta(hours=34, minutes=0).total_seconds() / 60
2040.0
>>> timedelta(hours=0, minutes=97).total_seconds() / 60
97.0
```

**Review by jtyoung84 — APPROVED** [review]

**Pull request merged**